lib/govuk_app_config/govuk_error/configuration.rb

```diff
@@ -12,12 +12,14 @@ def initialize(_raven_configuration)
       @data_sync = GovukDataSync.new(ENV["GOVUK_DATA_SYNC_PERIOD"])
       self.active_sentry_environments = []
       self.data_sync_excluded_exceptions = []
-      self.should_capture = ignore_exceptions_based_on_env_and_data_sync
+      self.before_send = lambda do |error_or_event, hint|
+        (ignore_exceptions_based_on_env_and_data_sync.call(error_or_event, hint) && increment_govuk_statsd_counters.call(error_or_event, hint))
+      end
     end
 
-    def should_capture=(closure)
-      combined = lambda do |error_or_event|
-        (ignore_exceptions_based_on_env_and_data_sync.call(error_or_event) && closure.call(error_or_event))
+    def before_send=(closure)
+      combined = lambda do |error_or_event, hint|
+        (ignore_exceptions_based_on_env_and_data_sync.call(error_or_event, hint) && increment_govuk_statsd_counters.call(error_or_event, hint) && closure.call(error_or_event, hint))
```

Comment:
I wonder if it's worth extracting the 'default' behaviour into its own method, so that we remove the duplication between lines 16 and 22. E.g.

```ruby
def before_send=(closure)
  combined = lambda do |error_or_event, hint|
    default_before_send_actions.call(error_or_event, hint) && closure.call(error_or_event, hint)
  end
  super(combined)
end
```

I wonder what happens now if we call `before_send=` 🤔  I can't remember if it stacks on top of previous calls, e.g.

```ruby
before_send = do_something
before_send = do_something_else

# when an error DOES happen, what then gets evaluated?
# is it:
[ default_before_send_actions, do_something, do_something_else ]
# or is it:
[ default_before_send_actions, do_something, default_before_send_actions, do_something, do_something_else ]
```

I hope it's the former, but if it's the latter, it hasn't been a problem until now, because it just evaluates whether or not to log the error. But now that _increments_ are being brought into the mix, we want to make sure that we're not double-counting. Does that make sense?
Would it be possible to add a test to check that behaviour?

Reply:
Good point — I don't _think_ it would stack (which is why the default is set in the constructor and then prepended in the setter method), but it can't hurt to have a test to ensure this.

Reply:
Looks like this is what's happening: because setting `before_send` in `initalize` ends up setting it to `default_before_send_actions && default_before_send_actions`.

[This](https://github.com/alphagov/govuk_app_config/pull/196/files#diff-59d0fe03fd43214b7f65eb27094e958f04c10366cbf2c5b5e3e470483899fc72R15) is a bit of a hack, but on the other hand, if `before_send` isn't initialised at all then the default actions won't be set either. I'm not sure if there's a better solution; I wondered about having an array of lambdas which `before_send` could call in turn, and which clients could add to if necessary, but that would mean altering the interface for dependants (and setting `before_send` directly would break it).

Reply:
Yes, we may want to make it explicit in documentation somewhere, though there's only one instance of `before_send` being used on GOV.UK and they've [explicitly returned the error](https://github.com/alphagov/govuk-coronavirus-vulnerable-people-form/blob/37fd197b2d94e55f3ed759b381181d282c4a18c8/config/application.rb#L33), so I'm hoping this is a non-problem :)